src/public/client.ts

```diff
@@ -168,19 +201,32 @@ class MappingItem {
   }
 }
 
-fetch('/api/mappings')
-  .then(r => r.json())
-  .then((data: Mapping[]) => {
-    domainList.innerHTML = ''
-    data.reverse()
-    data
-      .filter(
-        e => e.domain && e.port && e.id && e.ip && e.gitLink && e.fullDomain
-      )
-      .forEach(e => {
-        new MappingItem(e)
-      })
-  })
+Promise.all([
+  fetch('/api/mappings').then(r => r.json()),
+  fetch('/api/statuses').then(r => r.json())
+]).then(([mappings, statuses]: [Mapping[], Status[]]) => {
+  const fullDomainStatusMap = mappings.reduce((totalMap, dom) => {
+    if (!statuses[dom.fullDomain]) {
+      return [...totalMap, { ...dom, status: 'not started' }]
+    }
+
+    return [
+      ...totalMap,
+      {
+        ...dom,
+        status: statuses[dom.fullDomain]
+      }
+    ]
+  }, [])
+
+  domainList.innerHTML = ''
+  fullDomainStatusMap.reverse()
+  fullDomainStatusMap
+    .filter(e => e.domain && e.port && e.id && e.gitLink && e.fullDomain)
+    .forEach(e => {
+      new MappingItem(e)
```

Comment:
Could this accomplish the same thing, without as much code?

* remove `const fullDomainStatusMap = ...` and instead use `mappings.reverse().filter....`
```
domainList.innerHTML = ''
domainList
  .reverse()
  .filter(....)
  .forEach( e => {
    e.status = statuses[e.fullDomain]
    new MappingItem(e
  })
```